doc/guides/device_mgmt/index.rst

```diff
@@ -39,14 +39,417 @@ functionality over BLE and serial.
+   * - ``shell``
+     - Execute a command in the remote shell. This option is disabled by default
+       and can be enabled with :option:`CONFIG_MCUMGR_CMD_SHELL_MGMT` = ``y``.
+       To know more about the shell in Zephyr check :ref:`shell_api`.
+   * - ``stat``
+     - Read statistics from a device. More info in :ref:`stats_mgmt`.
+   * - ``taskstat``
+     - Read task statistics from a device. This command is part of the ``OS``
+       group which must be enabled by setting :option:`CONFIG_MCUMGR_CMD_OS_MGMT`.
+       The ``taskstat`` command itself can be enabled by setting
+       :option:`CONFIG_OS_MGMT_TASKSTAT`. :option:`CONFIG_THREAD_MONITOR` also
+       needs to be enabled otherwise a ``-8`` (``MGMT_ERR_ENOTSUP``) will be
+       returned.
+
+.. tip::
+
+    ``taskstat`` has a few options that might require tweaking. The
+    :option:`CONFIG_THREAD_NAME` must be set to display the task names, otherwise
+    the priority is displayed. Since the ``taskstat`` packets are large, they
+    might need increasing the :option:`CONFIG_MCUMGR_BUF_SIZE` option.
+
+.. warning::
+
+    To display the correct stack size in the ``taskstat`` command, the
+    :option:`CONFIG_THREAD_STACK_INFO` option must be set.
+    To display the correct stack usage in the ``taskstat`` command, both
+    :option:`CONFIG_THREAD_STACK_INFO` and :option:`CONFIG_INIT_STACKS` options
+    must be set.
+
+.. _image_mgmt:
+
+Image Management
+****************
+
+The image management provided by ``mcumgr`` is  based on the image format defined
+by MCUboot. For more details on the internals see `MCUboot design`_ and :ref:`west-sign`.
+
+To list available images in a device::
+
+  mcumgr <connection-options> image list
+
+This should result in an output similar to this::
+
+  $ mcumgr -c acm0 image list
+  Images:
+    image=0 slot=0
+      version: 1.0.0
+      bootable: true
+      flags: active confirmed
+      hash: 86dca73a3439112b310b5e033d811ec2df728d2264265f2046fced5a9ed00cc7
+  Split status: N/A (0)
+
+Where ``image`` is the number of the image pair in a multi-image system, and slot
+is the number of the slot where the image is stored, ``0`` for primary and ``1`` for
+secondary. This image being ``active`` and ``confirmed`` means it will run again on
+next reset. Also relevant is the ``hash``, which is used by other commands to
+refer to this specific image when performing operations.
+
+An image can be manually erased using::
+
+  mcumgr <connection-options> image erase
+
+The behavior of ``erase`` is defined by the server (``mcumgr`` in the device).
+The current implementation is limited to erasing the image in the secondary
+partition.
+
+To upload a new image::
+
+  mcumgr <connection-options> image upload [-n] [-e] [-u] <signed-bin>
+
+* ``-n``: This option allows uploading a new image to a specific set of images
+  in a multi-image system, and is currently only supported by MCUboot when the
+  ``BOOT_SERIAL`` option is enabled.
+
```

Comment:
Is it worth adding here a tip that you may check hash of the image you are uploading to the device with invocation of `image verify` on the binary?

Reply:
Do you mean `imgtool verify`?

Reply:
Yes, right.

Reply:
I honestly never use it :-)

The idea was to verify an image received with some key, but if you just did `image create` (or `sign`) I can't see no way this would fail. I will try to find some place to add the tip though.

Reply:
I use it to check what I have uploaded actually. The imgtool does not print sha when signing an image so the only way to get the ha is to verify the image after signing. If you test a lot of images, for some reason, you may get lost and sometimes need to check what is actually there.

Reply:
Done.